### PR TITLE
Adding the sensors and location compatibility

### PR DESCRIPTION
**Pull request by clementh59**

### CODE REVIEW

File: evrythng-proxy/EpcisEvent/epcisEvents.js [hunk 1]
```diff
@@ -0,0 +1,646 @@
...
+const lookupPlace = (app) => async (epc) => {
+  const identifier = epcToTuple(epc)[0];
+  const _epc = epcToTuple(epc)[1];
+  const place = await app.place().read({
+    params: {
+      filter: `${identifier}=${_epc}`,
+    },
+  });
+  if (place.length > 1) {
+    throw Error(`More than one entry found for ${epc}`);
```
Comment (joelvogt) [inline evrythng-proxy/EpcisEvent/epcisEvents.js:63]: Javascript also requires the new operator to create instances: throw new Error(`More...`)

File: evrythng-proxy/EpcisEvent/epcisEvents.js [hunk 2]
```diff
@@ -0,0 +1,646 @@
...
+    (
+      await Promise.all(
+        epcList
+          .map(epcToTuple)
+          .map((e) => e[1])
+          .map(lookupGS1Thng(app)),
+      )
+    ).join('').length !== 0
+  ) {
+    throw Error('Some thngs already exist. They cannot be added twice');
```
Comment (joelvogt) [inline evrythng-proxy/EpcisEvent/epcisEvents.js:254]: Same here, the new operator is missing

File: evrythng-proxy/EpcisEvent/epcisEvents.js [hunk 3]
```diff
@@ -0,0 +1,646 @@
...
+  productsGtins.forEach(async (gtin) => {
+    promises.push(async () => {
+      const product = await app.product().read({
+        params: {
+          filter: `identifiers.gs1:01=${gtin}`,
+        },
+      });
+
+      if (Object.keys(product).length === 0)
+        throw Error(`Invalid EPC  because product ${gtin} does not exist.`);
```
Comment (joelvogt) [inline evrythng-proxy/EpcisEvent/epcisEvents.js:312]: new Error

File: evrythng-proxy/EpcisEvent/epcisEvents.js [hunk 4]
```diff
@@ -0,0 +1,646 @@
...
+  let thngs = await Promise.all(
+    epcList
+      .map(epcToTuple)
+      .map((e) => e[1])
+      .map(createOrLookupGS1Thng(app)),
+  );
+  thngs = thngs.reduce((x, y) => x.concat(y)).map((t) => t.id);
+  print('deleteAction 3');
+  if (thngs.length !== epcList.length) {
+    throw Error('Could not find all the EPCs specified in the event');
```
Comment (joelvogt) [inline evrythng-proxy/EpcisEvent/epcisEvents.js:327]: new Error

File: evrythng-proxy/EpcisEvent/epcisEvents.js [hunk 5]
```diff
@@ -0,0 +1,646 @@
...
+  productsGtins.forEach(async (gtin) => {
+    promises.push(async () => {
+      const product = await app.product().read({
+        params: {
+          filter: `identifiers.gs1:01=${gtin}`,
+        },
+      });
+
+      if (Object.keys(product).length === 0)
+        throw Error(`Invalid EPC  because product ${gtin} does not exist.`);
```
Comment (joelvogt) [inline evrythng-proxy/EpcisEvent/epcisEvents.js:356]: new Error

File: evrythng-proxy/EpcisEvent/epcisEvents.js [hunk 6]
```diff
@@ -0,0 +1,646 @@
...
+
+  let thngs = await Promise.all(
+    epcList
+      .map(epcToTuple)
+      .map((e) => e[1])
+      .map(createOrLookupGS1Thng(app)),
+  );
+
+  if (thngs.length !== epcList.length) {
+    throw Error('Could not find all the EPCs specified in the event');
```
Comment (joelvogt) [inline evrythng-proxy/EpcisEvent/epcisEvents.js:370]: new Error

File: evrythng-proxy/EpcisEvent/epcisEvents.js [hunk 7]
```diff
@@ -0,0 +1,646 @@
...
+
+  /* eslint-disable class-methods-use-this */
+  /* eslint-disable no-unused-vars */
+  async deleteAction(epcList) {
+    throw Error('deleteAction not implemented');
+  }
+  /* eslint-disable class-methods-use-this */
+  /* eslint-disable no-unused-vars */
+
+  /* eslint-enable require-jsdoc */
```
Comment (joelvogt) [inline evrythng-proxy/EpcisEvent/epcisEvents.js:542]: Isn't jsdoc required by default?

**Review by joelvogt — CHANGES_REQUESTED** [review]

File: evrythng-proxy/EpcisEvent/epcisEvents.js (hunk 7 above)
Comment (clementh59) [reply evrythng-proxy/EpcisEvent/epcisEvents.js:542]: Yes but I disabled it here : https://github.com/evrythng/epcis2.0-to-evrythng/blob/clementHecquet-patch/evrythng-proxy/EpcisEvent/epcisEvents.js#L448

Comment (clementh59) [reply evrythng-proxy/EpcisEvent/epcisEvents.js:536]: Yes but I desibaled it here https://github.com/evrythng/epcis2.0-to-evrythng/blob/clementHecquet-patch/evrythng-proxy/EpcisEvent/epcisEvents.js#L442

**Review by joelvogt — APPROVED** [review]

**Pull request merged**